s3-sink-connector/src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -267,200 +164,19 @@ public void ensureValid(final String name, final Object value) {
-                LOGGER.info(AWS_S3_BUCKET + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, object);
-            }
-        }, Importance.MEDIUM, "AWS S3 Bucket name");
-
-        configDef.define(AWS_S3_ENDPOINT, Type.STRING, null, new UrlValidator() {
-            @Override
-            public void ensureValid(final String name, final Object object) {
-                LOGGER.info(AWS_S3_ENDPOINT + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, object);
-            }
-        }, Importance.LOW, "Explicit AWS S3 Endpoint Address, mainly for testing");
-
-        configDef.define(AWS_S3_REGION, Type.STRING, null, new AwsRegionValidator() {
-            @Override
-            public void ensureValid(final String name, final Object object) {
-                LOGGER.info(AWS_S3_REGION + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, object);
-            }
-        }, Importance.MEDIUM, "AWS S3 Region, e.g. us-east-1");
-
-        configDef.define(AWS_S3_PREFIX, Type.STRING, null, new ConfigDef.NonEmptyString() {
-            @Override
-            public void ensureValid(final String name, final Object object) {
-                LOGGER.info(AWS_S3_PREFIX + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, object);
-            }
-        }, Importance.MEDIUM, "Prefix for stored objects, e.g. cluster-1/");
-
-        configDef.define(OUTPUT_FIELDS, Type.LIST, null, new OutputFieldsValidator() {
-            @Override
-            public void ensureValid(final String name, final Object value) {
-                LOGGER.info(OUTPUT_FIELDS + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, value);
-            }
-        }, Importance.MEDIUM,
-                "Output fields. A comma separated list of one or more: " + OUTPUT_FIELD_NAME_KEY + ", "
-                        + OUTPUT_FIELD_NAME_OFFSET + ", " + OUTPUT_FIELD_NAME_TIMESTAMP + ", " + OUTPUT_FIELD_NAME_VALUE
-                        + ", " + OUTPUT_FIELD_NAME_HEADERS);
-
-        configDef.define(OUTPUT_COMPRESSION, Type.STRING, null, new FileCompressionTypeValidator() {
-            @Override
-            public void ensureValid(final String name, final Object value) {
-                LOGGER.info(OUTPUT_COMPRESSION + " property is deprecated please read documentation for the new name");
-                super.ensureValid(name, value);
-            }
-        }, Importance.MEDIUM, "Output compression. Valid values are: " + OUTPUT_COMPRESSION_TYPE_GZIP + " and "
-                + OUTPUT_COMPRESSION_TYPE_NONE);
+                timestampGroupCounter, ConfigDef.Width.SHORT, S3ConfigFragment.TIMESTAMP_SOURCE);
     }
 
     private void validate() {
```

Comment:
is it possible to move this validate method to common config? as this will be used by source class too.

Reply:
I think I can move the crednetial validation into the S3 Config Fragment to make it available to both S3Sink and Source. Will do that now.